src/main/java/com/google/api/codegen/util/SymbolTable.java

```diff
@@ -17,23 +17,59 @@
 import java.util.HashSet;
 import java.util.Set;
 
+import com.google.common.base.Strings;
+
 /**
  * A utility class used to get and store unique symbols.
  *
  * If a symbol is already used, the table will try to append an index number onto the end of it.
  * The index will keep increasing until an unused symbol is found.
  */
 public class SymbolTable {
+
   private final Set<String> symbolTable = new HashSet<>();
 
+  /**
+   * Returns a unique name, with a numeric suffix in case of conflicts.
+   *
+   * Not guaranteed to work as expected if used in combination with {@link
+   * #getNewSymbol(String)}.
+   */
   public Name getNewSymbol(Name desiredName) {
-    Name actualName = desiredName;
+    String lower = desiredName.toLowerUnderscore();
+    String suffix = getAndSaveSuffix(lower);
+    if (Strings.isNullOrEmpty(suffix)) {
+      return desiredName;
+    }
+    return desiredName.join(suffix);
+  }
+
+  /**
+   * Returns a unique name, with a numeric suffix in case of conflicts.
+   *
+   * Not guaranteed to work as expected if used in combination with {@link
+   * #getNewSymbol(Name)}.
+   */
+  public String getNewSymbol(String desiredName) {
+    String suffix = getAndSaveSuffix(desiredName);
+    return desiredName + suffix;
+  }
+
+  /**
+   *  Returns the next numeric suffix that makes desiredName unique.
+   *
+   *  Stores the joined desiredName/suffix in an internal map.
+   */
+  private String getAndSaveSuffix(String desiredName) {
+    if (!symbolTable.contains(desiredName)) {
+      symbolTable.add(desiredName);
+      return "";
+    }
     int i = 2;
```

Comment:
Idea is:
`getSymbol("foo")` -> `"foo"`
`getSymbol("foo")` -> `"foo2"`

I'll make a comment to explain